### PR TITLE
[Feature] Add result information for analyze stmt and add general filter code in ShowExecutor

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 59 / 118 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|6|00.00%|[812, 813, 814, 815, 816, 819]|
|:large_blue_circle:|com/starrocks/qe/ShowResultSetMetaData.java|0|4|00.00%|[47, 48, 49, 52]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeStatus.java|7|29|24.14%|[142, 143, 145, 146, 149, 150, 152, 153, 154, 157, 158, 159, 160, 161, 166, 167, 168, 170, 171, 174, 175, 176]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|20|25.00%|[1560, 1576, 1591, 1607, 1638, 1639, 1640, 1641, 1643, 1644, 1645, 1646, 1647, 1649, 1652]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|11|45.45%|[335, 336, 339, 340, 341, 343]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|11|14|78.57%|[1204, 1220, 1230]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|4|5|80.00%|[98]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|4|5|80.00%|[71]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|4|5|80.00%|[71]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/ShowStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|4|4|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 1]
```diff
@@ -809,12 +809,14 @@ private void handleAnalyzeStmt() {
...
-        if (analyzeStatus.getStatus().equals(StatsConstants.ScheduleStatus.FINISH)) {
-            context.getState().setOk();
-        } else {
-            context.getState().setError(analyzeStatus.getReason());
+        ShowResultSet resultSet = analyzeStatus.toShowResult();
+        if (isProxy) {
+            proxyResultSet = resultSet;
+            context.getState().setEof();
+            return;
         }
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:817]: What's this?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:817]: This is the output information of the executed statement. If it is a follower, it needs to be forwarded.

File: fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeStatus.java [hunk 2]
```diff
@@ -119,6 +129,53 @@ public void setReason(String reason) {
         this.reason = reason;
     }
 
+    static final ShowResultSetMetaData META_DATA =
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeStatus.java:132]:
```suggestion
    private static final ShowResultSetMetaData META_DATA =
```

File: fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeStatus.java [hunk 3]
```diff
@@ -119,6 +129,53 @@ public void setReason(String reason) {
...
+            Table table = db.getTable(tableId);
+            tableName = table.getName();
+        }
+
+        String op = "unknown";
+        if (type.equals(StatsConstants.AnalyzeType.HISTOGRAM)) {
+            op = "histogram";
+        } else if (type.equals(StatsConstants.AnalyzeType.FULL) || type.equals(StatsConstants.AnalyzeType.SAMPLE)) {
+            op = "analyze";
+        }
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/statistic/AnalyzeStatus.java:162]: show full or sample is better?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 66 / 127 (51.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|6|00.00%|[812, 813, 814, 815, 816, 819]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|0|1|00.00%|[75]|
|:large_blue_circle:|com/starrocks/qe/ShowResultSetMetaData.java|0|4|00.00%|[47, 48, 49, 52]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeStatus.java|7|29|24.14%|[142, 143, 145, 146, 149, 150, 152, 153, 154, 157, 158, 159, 160, 161, 166, 167, 168, 170, 171, 174, 175, 176]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|20|25.00%|[1560, 1576, 1591, 1607, 1638, 1639, 1640, 1641, 1643, 1644, 1645, 1646, 1647, 1649, 1652]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|11|45.45%|[335, 336, 339, 340, 341, 343]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|11|14|78.57%|[1204, 1220, 1230]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|4|5|80.00%|[71]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|4|5|80.00%|[71]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|11|13|84.62%|[72, 103]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/ShowStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 65 / 128 (50.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|6|00.00%|[812, 813, 814, 815, 816, 819]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsCollectJobFactory.java|0|1|00.00%|[75]|
|:large_blue_circle:|com/starrocks/qe/ShowResultSetMetaData.java|0|4|00.00%|[47, 48, 49, 52]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeStatus.java|6|30|20.00%|[141, 142, 144, 145, 148, 149, 151, 152, 153, 156, 157, 158, 159, 160, 161, 162, 167, 168, 169, 171, 172, 175, 176, 177]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|5|20|25.00%|[1560, 1576, 1591, 1607, 1638, 1639, 1640, 1641, 1643, 1644, 1645, 1646, 1647, 1649, 1652]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|11|45.45%|[335, 336, 339, 340, 341, 343]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|11|14|78.57%|[1204, 1220, 1230]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowBasicStatsMetaStmt.java|4|5|80.00%|[71]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowHistogramStatsMetaStmt.java|4|5|80.00%|[71]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeStatusStmt.java|11|13|84.62%|[72, 103]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/ShowStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowAnalyzeJobStmt.java|4|4|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**